OpenSSL/_util.py

```diff
@@ -68,6 +70,23 @@ def native(s):
 
 
 
+def path_string(s):
+    """
+    Convert a Python string to a :py:class:`bytes` string identifying the same
+    path and which can be passed into an OpenSSL API accepting a filename.
+
+    :param s: An instance of :py:class:`bytes` or :py:class:`unicode`.
+
+    :return: An instance of :py:class:`bytes`.
+    """
+    if isinstance(s, binary_type):
+        return s
+    elif isinstance(s, text_type):
+        return s.encode(sys.getfilesystemencoding())
```

Comment:
Looking over #62 again, I think `return filename.encode(sys.getfilesystemencoding() or 'ascii')` is the best option until we can divinate `open` more.  Just using `.encode()` with no argument is probably equivalent to `.encode(sys.getdefaultencoding())` which you didn't think was the right way (I'm not really sure).

With out the `or 'ascii'` Py2 users may get a `TypeError: encode() argument 1 must be string, not None` which will be rather cryptic.